lib/cocoapods/validator.rb

```diff
@@ -921,6 +921,14 @@ def xcodebuild(action, scheme, configuration)
       when :ios
         command += %w(CODE_SIGN_IDENTITY=- -sdk iphonesimulator)
         command += Fourflusher::SimControl.new.destination(:oldest, 'iOS', deployment_target)
+        xcconfig = consumer.spec.attributes_hash['pod_target_xcconfig']
```

Comment:
You want to use `consumer.pod_target_xcconfig` here as it will take care of merging values etc. See https://github.com/CocoaPods/Core/blob/master/lib/cocoapods-core/specification/consumer.rb#L99-L102